tools/kubernetes/migration.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2025 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import requests
+import yaml
+import os
+from pathlib import Path
+from kfp.dsl import utils
+
+# Constants and defaults
+K8S_NAMESPACE = 'kubeflow'
+K8S_PIPELINE_API_VERSION = 'pipelines.kubeflow.org/v2beta1'
+MAX_NAME_LENGTH = 253
+DEFAULT_NAMESPACE = 'kubeflow'
+DEFAULT_OUTPUT_DIR = './kfp-exported-pipelines'
+DEFAULT_TIMEOUT = 10  # seconds
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Migrate KFP pipelines to Kubernetes manifests")
+    parser.add_argument("--kfp-server-host", default=os.getenv("KFP_SERVER_HOST"), required=True,
+                        help="KFP pipeline server host (e.g., https://<host>)")
+    parser.add_argument("--token", default=os.getenv("KFP_BEARER_TOKEN"), help="Bearer token for authentication")    
+    parser.add_argument("--ca-bundle", default=os.getenv("CA_BUNDLE"), help="Path to custom CA bundle file")
+    parser.add_argument('--output', '-o', default=DEFAULT_OUTPUT_DIR, help="Output directory path where pipeline YAMLs will be written(e.g., '.path/to/exported-pipelines')")
+    parser.add_argument('--namespace', default=DEFAULT_NAMESPACE, help="Namespace to filter pipelines from")
+    parser.add_argument('--batch-size', type=int, default=100,
```

Comment:
Since we are using pagination, we can lower this default to something smaller like `20` or `50` results.